drift-codec/src/main/java/io/airlift/drift/codec/internal/EnumThriftCodec.java

```diff
@@ -57,12 +57,8 @@ public T read(TProtocolReader protocol)
         if (enumConstant != null) {
             return enumConstant;
         }
-        // unknown, throw unknown value exception
-        throw new UnknownEnumValueException(
-                String.format(
-                        "Enum %s does not have a value for %s",
-                        enumMetadata.getEnumClass(),
-                        enumValue));
+        return enumMetadata.getUnknownEnumConstant()
+                .orElseThrow(() -> new UnknownEnumValueException(String.format("Enum %s does not have a value for %s", enumMetadata.getEnumClass(), enumValue)));
```

Comment:
Static import

Reply:
Use `enumMetadata.getEnumClass().getName()`, unless you wanted it to say `"Enum class XXX"`

Reply:
Maybe change the message to
```java
"Enum %s does not have a constant for value: %s"